routes/search/router.js

```diff
@@ -77,20 +97,50 @@ router.get('/', function(req, res, next) {
             let distances = response.json.rows[0].elements;
             distances.forEach(function(el, i) {
               if (el.distance) {
-                results[i].distance = el.distance.value;
-                results[i].duration = el.duration.value;
+                freelancers[i].distance = el.distance.value;
+                freelancers[i].duration = el.duration.value;
               } else {
-                results[i].distance = Number.MAX_SAFE_INTEGER;
-                results[i].duration = Number.MAX_SAFE_INTEGER;
+                freelancers[i].distance = Number.MAX_SAFE_INTEGER;
+                freelancers[i].duration = Number.MAX_SAFE_INTEGER;
               }
             });
-            res.json(results).end();
+            googleMapsClient.distanceMatrix({
+              origins: [ req.query.origin ],
+              destinations: destinationsEmergency
+            }, function(err, response) {
+              let distancesEmergency = response.json.rows[0].elements;
+              distancesEmergency.forEach(function(r, i) {
+                if (r.distance) {
+                  freelancers[i].emergency.distance = r.distance.value;
+                  freelancers[i].emergency.duration = r.duration.value;
+                } else {
+                  freelancers[i].emergency.distance = Number.MAX_SAFE_INTEGER;
+                  freelancers[i].emergency.duration = Number.MAX_SAFE_INTEGER;
+                }
+              });
+              res.json(freelancers).end();
+            });
           }
         });
       }
     });
   });
 });
 
+// Check if a freelancer is available in a given moment
+function freelancerAvailableDay(freelancer, dateStr) {
+  let date = new Date(dateStr);
```

Comment:
The Date of the search query should be changed to UTC, otherwise check will be wrong (i.e. if I search for emergency now it will not show freelancers available now, but only freelancers available in two hours because the availabilities are stored as UTC)